cmd/kusk/cmd/uninstall.go

```diff
@@ -0,0 +1,188 @@
+package cmd
+
+import (
+	"fmt"
+	"path/filepath"
+
+	kuskv1 "github.com/kubeshop/kusk-gateway/api/v1alpha1"
+	"github.com/kubeshop/kusk-gateway/cmd/kusk/internal/errors"
+	"github.com/kubeshop/kusk-gateway/cmd/kusk/internal/utils"
+	"github.com/manifoldco/promptui"
+	"github.com/spf13/cobra"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var confirm bool
+
+var uninstallCmd = &cobra.Command{
+	Use:   "uninstall",
+	Short: "Uninstall kusk-gateway, envoy-fleet, api, and dashboard in a single command",
+	Long: `
+	Uninstall kusk-gateway, envoy-fleet, api, and dashboard in a single command.
+
+	$ kusk uninstall
+
+	Will install kusk-gateway, a public (for your APIS) and private (for the kusk dashboard and api)
+	envoy-fleet, api, and dashboard in the kusk-system namespace using helm.
+
+	$ kusk install --name=my-release --namespace=my-namespace
+
+	Will create a helm release named with --name in the namespace specified by --namespace.
+
+	$ kusk install --no-dashboard --no-api --no-envoy-fleet
+
+	Will install kusk-gateway, but not the dashboard, api, or envoy-fleet.
+	`,
+	SilenceUsage:  true,
+	SilenceErrors: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		reportError := func(err error) {
+			if err != nil {
+				errors.NewErrorReporter(cmd, err).Report()
+			}
+		}
+
+		proceed := confirm
+
+		if !confirm {
+			prompt := promptui.Prompt{
+				Label:     "uninstall command is irrevirsible. Are you sure you want to proceed?",
+				IsConfirm: true,
+			}
+			result, err := prompt.Run()
+			if err != nil {
+				return nil
+			}
+
+			if result == "N" || result == "n" || result == "" {
+				fmt.Println("exiting")
+				return nil
+			}
+			proceed = true
+		}
+
+		if proceed {
```

Comment:
if !proceed then just return early

Reply:
there is a prompt that sets proceed to true. Probably i over-complicated it

Reply:
I'm disabling prompt if --no-confirm is set. So if proceed is false we want to prompt the user with y/N,